docs/elasticsearch-spec.asciidoc

```diff
@@ -359,64 +359,51 @@ The init container inherits the image of the main container image if one is not
+* `null` - the default value will be used,
+* non-negative - the value will be used as is,
+* negative - the value will be unbounded.
 
-This behaviour can be controlled through the `changeBudget` section of the cluster specification `updateStrategy`. If not specified, it defaults to the following:
+==== Default behavior
+When `updateStrategy` is not present in the spec, it defaults to the following:
 
 [source,yaml]
 ----
 spec:
   updateStrategy:
     changeBudget:
-      maxSurge: 1
-      maxUnavailable: 0
+      maxSurge: -1
+      maxUnavailable: 1
 ----
+`maxSurge` is unbounded. This results in immediate creation of all missing Pods that best practices allow for at the given time.
+`maxUnavailable` defaults to `1`. This assures that cluster will have at most one unavailable Pod at any given time.
 
-* `maxSurge` specifies the number of Pods that can be added to the cluster, on top of the desired number of nodes in the specification during cluster updates
-* `maxUnavailable` specifies the number of Pods that can be made unavailable during cluster updates
-
-The default of `maxSurge: 1; maxUnavailable: 0` spins up an additional Elasticsearch node during cluster updates.
-It is possible to speed up cluster topology changes by increasing `maxSurge`. For example, setting `maxSurge: 3` would allow 3 new nodes to be created while the original 3 migrate data in parallel.
-The cluster would then temporarily have 6 nodes.
-
-Setting `maxSurge` to 0 and `maxUnavailable` to a positive value only allows a maximum number of Pods to exist on the Kubernetes cluster.
-For example, `maxSurge: 0; maxUnavailable: 1` would perform the 3 nodes upgrade this way:
-
-1. Migrate data away from the first 16GB node
-2. Once data is migrated, remove the 16GB node: the cluster temporarily has 2 nodes
-3. Add a new 32GB node: the cluster grows to 3 nodes
-4. Follow the same steps for the 2 other 16GB nodes
-
-Even if a `changeBudget` is specified, ECK makes sure that some invariants are maintained while a mutation is in progress. In the cluster, there must be at least:
-
-* One master node alive
-* One data node alive
 
-So under certain circumstances ECK ignores the change budget. For example, a safe migration from a 1-node cluster to another 1-node cluster can only be done by temporarily setting up a 2-nodes cluster.
+==== Caveats
+* Setting both `maxSurge` and `maxUnavailable` to `0` will prevent the operator from performing any update that needs to bring down or create a new Pod.
+* Due to the safety measures employed by the operator, a particular changeBudget might be insufficient to perform the requested update. For example, removing last data node from one nodeSet and adding data node to a different nodeSet is not possible with `maxSurge` set to 0. In this case operator can't remove the old node (because it can't be migrated since no other data nodes are available) and can't add the new node (because `maxSurge` is set to 0).
```

Comment:
Due to the safety measures employed by the operator, a particular changeBudget might be insufficient to perform the requested update. For example, with maxSurge set to 0 you cannot remove the last data node from one nodeSet and add a data node to a different nodeSet. In this case the operator cannot remove the old node and migrate data as other nodes are not available, and can't add the new node because maxSurge is set to 0.